awx/main/dispatch/pool.py

```diff
@@ -401,13 +401,15 @@ def cleanup(self):
                 # the task manager to never do more work
                 current_task = w.current_task
                 if current_task and isinstance(current_task, dict):
-                    if current_task.get('task', '').endswith('tasks.run_task_manager'):
+                    endings = ['tasks.task_manager', 'tasks.dependency_manager', 'tasks.workflow_manager']
+                    current_task_name = current_task.get('task', '')
+                    if any([current_task_name.endswith(e) for e in endings]):
```

Comment:
FYI, nesting a list comprehension in like this isn't necessary.  `any(current_task_name.endswith(e) for e in endings)` will work just fine.

Reply:
nice catch, fixed